src/encoded/util.py

```diff
@@ -245,15 +257,86 @@ def s3_input_stream(s3_client, bucket: str, key: str, mode: str = 'r'):
             yield fp
 
 
-def create_empty_s3_file(s3_client, bucket: str, key: str):
+@contextlib.contextmanager
+def temporary_file(*, extension: Optional[str] = None) -> Generator[Any, None, None]:
+    """
+    Context manager allowing something like this:
+        with temporary_file(extension=".json") as filename:
+            with io.open(filename, "w") as fp:
+                put_something_in_the_temporary_file(fp)
+            with io.open(filename, "r") as fp:
+                process_the_temporary_file(fp)
+    :param extension: Optional filename extension (with the dot prefix if desired, e.g. ".json").
+    """
+    temporary_filename = tempfile.mktemp() + (extension if extension else "")
+    try:
+        yield temporary_filename
+    finally:
+        try:
+            os.remove(temporary_filename)
+        except Exception:
+            pass
+
+
+class SettingsKey:
+    APPLICATION_BUCKET_PREFIX = 'application_bucket_prefix'
+    BLOB_BUCKET = 'blob_bucket'
+    EB_APP_VERSION = 'eb_app_version'
+    ELASTICSEARCH_SERVER = 'elasticsearch.server'
+    ENCODED_VERSION = 'encoded_version'
+    FILE_UPLOAD_BUCKET = 'file_upload_bucket'
+    FILE_WFOUT_BUCKET = 'file_wfout_bucket'
+    FOURSIGHT_BUCKET_PREFIX = 'foursight_bucket_prefix'
+    IDENTITY = 'identity'
+    INDEXER = 'indexer'
+    INDEXER_NAMESPACE = 'indexer.namespace'
+    INDEX_SERVER = 'index_server'
+    LOAD_TEST_DATA = 'load_test_data'
+    METADATA_BUNDLES_BUCKET = 'metadata_bundles_bucket'
+    S3_ENCRYPT_KEY_ID = 's3_encrypt_key_id'
+    SNOVAULT_VERSION = 'snovault_version'
+    SQLALCHEMY_URL = 'sqlalchemy.url'
+    SYSTEM_BUCKET = 'system_bucket'
+    TIBANNA_CWLS_BUCKET = 'tibanna_cwls_bucket'
+    TIBANNA_OUTPUT_BUCKET = 'tibanna_output_bucket'
+    UTILS_VERSION = 'utils_version'
+
+
+class ExtraArgs:
+    SERVER_SIDE_ENCRYPTION = "ServerSideEncryption"
+    SSE_KMS_KEY_ID = "SSEKMSKeyId"
+
+
+def extra_kwargs_for_s3_encrypt_key_id(s3_encrypt_key_id, client_name):
+
+    extra_kwargs = {}
+    if s3_encrypt_key_id:
+        log.error(f"{client_name} adding SSEKMSKeyId ({s3_encrypt_key_id}) arguments in upload_fileobj call.")
+        extra_kwargs["ExtraArgs"] = {
+            ExtraArgs.SERVER_SIDE_ENCRYPTION: "aws:kms",
+            ExtraArgs.SSE_KMS_KEY_ID: s3_encrypt_key_id,
+        }
+    else:
+        log.error(f"{client_name} found no s3 encrypt key id ({SettingsKey.S3_ENCRYPT_KEY_ID})"
```

Comment:
These two can probably be downgraded to warning especially since there is no `S3_ENCRYPT_KEY_ID ` in fourfront.